### ISSUE TITLE
Collection drag and drop doesn't work properly

### ISSUE THREAD

**Issue by AbdelrahmanHafez** [a85/POSTMan-Chrome-Extension#4120]
I've been trying to recover my lost requests for the past 4 hours, and I found a backup.
The issue is that when I tried to move a folder from one level to another, or even reorder it in the same level. Very weird things have happened, requests disappeared, even some folders.

The surprising solution was that I use the light theme instead of the dark theme, I don't know if this can be relevant or not, but things seem to be working much more smoothly when I switched to the light theme.

Postman for Windows
Version 5.5.2
win32 10.0.10586 / x64

**Comment by Iliev88** [comment]
I am always using dark theme and such kind of problem is not faced.

**Comment by AbdelrahmanHafez** [comment]
Do you happen to be syncing with a team? It's possible this bug has to do with syncing.

**Comment by prashantagarwal** [comment]
@AbdelrahmanHafez We will be releasing some improvements to drag and drop in upcoming release. This will probably get fixed with those enhancements.

**Comment by allyn-psyonix** [comment]
Just wanted to chime in: About 80% of the time that I attempt to drag-and-drop move an entry within a collection it just disappears. The tab with the service call etc is still open, so it seems to still exist somewhere, but I have manually searched all collections and cannot find any of the missing tests.

I am using the standalone client, dark theme, Windows 10, and we are using shared team collections.

This is particularly infuriating because the order of tests matters for our automated build pipeline.

**Comment by allyn-psyonix** [comment]
Update: Someone suggesting the theme might be involved, but I tested with the light theme and the same problem occurs (even after restarting postman)

**Comment by allyn-psyonix** [comment]
Update 2: It looks like the disappearing tests are still in the collection, they just don't appear in the postman client. They are getting run when we run the full collection via Newman, except now I cannot delete them using the postman client.

**Comment by allyn-psyonix** [comment]
Update 3: Looks like other users can see the "disappearing" postman tests in their client, but a resync and a client restart still does not make them appear on the client that was used to move them.

I realize there was a post indicating this might be fixed by the drag and drop improvements in the upcoming release, but I wanted to provide my findings to help test it in case the updates don't end up fixing this issue directly.

**Comment by AbdelrahmanHafez** [comment]
I believe it's more related to having requests in different order on multiple devices with you & your team, one way I approached this is that I created a backup, deleted the entire collection, re-invited my whole team again.

I noticed that things get better for drag & drop when you wait for the thing you dragged & dropped / created / duplicated to sync before doing anything else at all.
Also, never move anything two levels up/down, always take it one level at a time. Using these two rules solved the issue for me till it's fixed.

Also, you could use the GitLab integration, that creates a commit with each change you make to the collection, in case things go terribly, you can know that you'll still have backups.

**Comment by sdnts** [comment]
@allyn-psyonix @AbdelrahmanHafez We've made improvements to both drag and drop, as well as Sync. These will drop on our stable channels soon (v6.0). Can you guys retry then to check if the issue still persists?

**Comment by allyn-psyonix** [comment]
@madebysid Absolutely. Is there a way to get notified when the 6.0 release happens? Or an estimated timeline so I know when to check back?

**Comment by AbdelrahmanHafez** [comment]
Sure! Would appreciate it if you leave a comment here so we get notified about it when v6.0 is released.

**Comment by sdnts** [comment]
Sure. You should also receive an update notification in your apps.

**Comment by allyn-psyonix** [comment]
I just tried 6.0.8 and the postman tests that were not syncing to 5.x properly are still not syncing to 6.0.8 properly. Tests appear out of order and some are missing, but they all run in the correct order when run through Newman. You can also see the missing tests in the chrome plugin.

**Comment by allyn-psyonix** [comment]
Update: Things seem to have gotten worse with 6.0.8. Previously, moving tests around would cause them to disappear permanently from the standalone app about 85% of the time, but I never had an issue with "save as" working properly. Now, I just tried "save as" and the new copy does not appear in my standalone client. It does appear in the chrome plugin though. I never experienced this issue with 5.x.

**Comment by allyn-psyonix** [comment]
Can we get this changed back to open? It's definitely still a serious issue in 6.x (I am currently on 6.0.9). I just duplicated an item and the new copy does not appear in the list. This is the third or fourth different type of action that causes items to disappear. Manual resync from settings does not resolve the issue, only manually deleting the local file system cache and restarting the app gets it back into a good state.

**Comment by prashantagarwal** [comment]
@allyn-psyonix Can you help me with the exact steps to reproduce this issue. Also were you using sidebar filter while duplicating the request. Thanks

**Comment by allyn-psyonix** [comment]
To reproduce, do any of the following in the list along the left side:
- Drag and drop a test to another folder. It can be within the same collection or a new collection.
- Drag and drop a folder to a new collection
- Choose an item and select "Duplicate"

All of these have about an 80% chance for the item or duplicate to disappear from the list. For the second one (dragging a whole folder), sometimes the items in the folder get duplicated and the order gets jumbled, and this can happen regardless of whether or not the folder disappears. If it also disappears, you won't see the duplicates until you do the steps below to recover. I have not been able to identify anything to indicate why the problem only occurs most of the time but not every single time.

A manual resync from the settings screen does not fix the issue, the item(s) will still not be visible. To resyc properly and see the disappeared items again you must close Postman and delete the "Cache" folder on the local disk. Note that the runner only runs the visible tests in Postman, but Newman sees all the tests and runs everything.

No filters are active in the list on the left.

This problem affects the whole team and has been an issue in both 5.x and 6.0.8/9. I don't recall the earliest 5.x version affected.

**Comment by allyn-psyonix** [comment]
In case it's relevant, we have quite a few tests. We have six shared collections with the largest two containing over 800 tests between them. Most of those are in folders within their respective collections.

I also just noticed that 6.0.10 is out so I updated and dragged a few tests around and do not see the issue at the moment. I'm hesitant to say everything is working fine since the change list doesn't look like anything related to sync, but typically if I drag 5 items I would normally expect a few to disappear.
